### ISSUE TITLE
`svelte/no-unused-props` false positives

### ISSUE THREAD

**Issue by benmccann** [sveltejs/eslint-plugin-svelte#1155]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.18.0

### What version of `eslint-plugin-svelte` are you using?

3.3.3

### What did you do?

Upgraded https://github.com/immich-app/immich/tree/main/web to `eslint-plugin-svelte` and removed the props reported as unused. Tried to send a PR just to remove those props without the `eslint-plugin-svelte` upgrade and found they weren't actually unused: https://github.com/immich-app/immich/pull/17120

### What did you expect to happen?

Should not report props as unused when they're actually used

### What actually happened?

Reported props as unused when they were actually used

E.g.:
```
src/lib/components/asset-viewer/photo-viewer.svelte
  38:7  error  'onClose' is an unused Props property  svelte/no-unused-props
```

But it's used here:
https://github.com/immich-app/immich/blob/dfb0626c91f2ec8f2a1887b14ec5995024cb1a73/web/src/lib/components/asset-viewer/asset-viewer.svelte#L485

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/immich-app/immich/tree/main/web

(can disable the TypeScript rules to speed up linting)

**Comment by baseballyama** [comment]
This rule reports properties that are defined in the type but not used via `$props`.
In this case, `onClose` is defined in the type but is actually unused.

https://github.com/immich-app/immich/blob/dfb0626c91f2ec8f2a1887b14ec5995024cb1a73/web/src/lib/components/asset-viewer/photo-viewer.svelte#L25-L48

(Unused variables declared via `$props` can be detected by `no-unused-var`.)

**Comment by benmccann** [comment]
ohhh, thanks! I was looking at the callers, but not where it was used 😆

**Comment by benmccann** [comment]
Part of the reason I was confused is that it turns out you don't get a warning when passing a prop that doesn't exist: https://github.com/sveltejs/eslint-plugin-svelte/issues/1156
